tests/compiler/resolve-typeof.ts

```diff
@@ -0,0 +1,4 @@
+(typeof 1).length;
+
+function generic<T>(v: T): void {}
+generic(typeof ((): void => {}));
```

Comment:
This would fit into the `compiler/resolve-unary` test I think, so we don't need a separate file :)